spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpMessageMapper.java

```diff
@@ -132,6 +132,10 @@ protected final void addStandardHeaders(TcpConnection connection, AbstractIntegr
 			.setHeader(IpHeaders.IP_ADDRESS, connection.getHostAddress())
 			.setHeader(IpHeaders.REMOTE_PORT, connection.getPort())
 			.setHeader(IpHeaders.CONNECTION_ID, connectionId);
+		SocketInfo socketInfo = connection.getSocketInfo();
+		if (socketInfo != null) {
+			messageBuilder.setHeader(IpHeaders.LOCAL_ADDRESS, socketInfo.getLocalAddress());
```

Comment:
Yes; I'd rather not carry a live reference to the socket in messages by default - the whole thing will be dropped if serialized anyway. In custom mappers, I expect users to only populate the results of the getters on `SocketInfo` - not the whole thing.